gossip/src/duplicate_shred_handler.rs

```diff
@@ -139,21 +172,31 @@ impl DuplicateShredHandler {
                 return false;
             }
         }
-        // Also skip frozen slots or slots with an older proof than me.
+        // Also skip frozen slots or slots with a different proof than me.
         match self.chunk_map.get(&slot) {
-            Some(SlotStatus::Frozen) => {
-                return false;
-            }
-            Some(SlotStatus::UnfinishedProof(slot_map)) => {
-                if let Some(proof_chunkmap) = slot_map.peek(&data.from) {
-                    if proof_chunkmap.wallclock < data.wallclock {
-                        return false;
-                    }
-                }
+            Some(SlotStatus::Frozen) => false,
+            Some(SlotStatus::UnfinishedProof(slot_map)) => match slot_map.get(&data.from) {
+                None => true,
+                Some(proof_chunkmap) => proof_chunkmap.wallclock == data.wallclock,
+            },
+            None => true,
+        }
+    }
+
+    fn dump_pubkeys_with_low_stakes(
+        cached_staked_nodes: Arc<HashMap<Pubkey, u64>>,
```

Comment:
this can be a just be `&HashMap<...>`;
does not need to be an `Arc`

Reply:
done